pkg/runtime/framework/plugins/coscheduling/coscheduling_test.go

```diff
@@ -0,0 +1,326 @@
+					},
+				},
+				RuntimePolicy: runtime.RuntimePolicy{
+					PodGroupPolicy: &trainerv1alpha1.PodGroupPolicy{
+						PodGroupPolicySource: trainerv1alpha1.PodGroupPolicySource{
+							Coscheduling: &trainerv1alpha1.CoschedulingPodGroupPolicySource{
+								ScheduleTimeoutSeconds: ptr.To[int32](30),
+							},
+						},
+					},
+				},
+				TemplateSpec: runtime.TemplateSpec{
+					PodSets: []runtime.PodSet{
+						{
+							Name:  "node",
+							Count: ptr.To[int32](1),
+						},
+					},
+				},
+			},
+			objs: []client.Object{}, // Simulate no existing PodGroup
+			wantObjs: []apiruntime.Object{
+				utiltesting.MakeSchedulerPluginsPodGroup(metav1.NamespaceDefault, "trainJob").
+					MinMember(1).
+					MinResources(corev1.ResourceList{}).
+					SchedulingTimeout(30).
+					ControllerReference(trainerv1alpha1.GroupVersion.WithKind(trainerv1alpha1.TrainJobKind), "trainJob", "trainJob").
+					Obj(),
+			},
+		},
+		"failed to get PodGroup due to API error": {
+			info: &runtime.Info{
+				Scheduler: &runtime.Scheduler{},
+				Labels:    map[string]string{"key": "value"},
+				RuntimePolicy: runtime.RuntimePolicy{
+					PodGroupPolicy: &trainerv1alpha1.PodGroupPolicy{
+						PodGroupPolicySource: trainerv1alpha1.PodGroupPolicySource{
+							Coscheduling: &trainerv1alpha1.CoschedulingPodGroupPolicySource{
+								ScheduleTimeoutSeconds: ptr.To[int32](30),
+							},
+						},
+					},
+				},
+			},
+			trainJob: &trainerv1alpha1.TrainJob{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "trainJob",
+					Namespace: metav1.NamespaceDefault,
+				},
+			},
+			wantInfo: &runtime.Info{
+				Scheduler: &runtime.Scheduler{
+					PodLabels: map[string]string{
+						"scheduling.x-k8s.io/pod-group": "trainJob",
+					},
+				},
+				Labels: map[string]string{"key": "value"},
+				RuntimePolicy: runtime.RuntimePolicy{
+					PodGroupPolicy: &trainerv1alpha1.PodGroupPolicy{
+						PodGroupPolicySource: trainerv1alpha1.PodGroupPolicySource{
+							Coscheduling: &trainerv1alpha1.CoschedulingPodGroupPolicySource{
+								ScheduleTimeoutSeconds: ptr.To[int32](30),
+							},
+						},
+					},
+				},
+			},
+			wantBuildError: errorGetPodGroup,
+		},
+		"no action when PodGroup already exists": {
+			info: &runtime.Info{
+				Scheduler: &runtime.Scheduler{},
+				Labels:    map[string]string{"key": "value"},
+				RuntimePolicy: runtime.RuntimePolicy{
+					PodGroupPolicy: &trainerv1alpha1.PodGroupPolicy{
+						PodGroupPolicySource: trainerv1alpha1.PodGroupPolicySource{
+							Coscheduling: &trainerv1alpha1.CoschedulingPodGroupPolicySource{
+								ScheduleTimeoutSeconds: ptr.To[int32](30),
+							},
+						},
+					},
+				},
+			},
+			trainJob: &trainerv1alpha1.TrainJob{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "trainJob",
+					Namespace: metav1.NamespaceDefault,
+				},
+				Spec: trainerv1alpha1.TrainJobSpec{
+					Suspend: ptr.To(false),
+				},
+			},
+			objs: []client.Object{
```

Comment:
Please add a tests case when TrainJob is suspended.
Plugin should not return any objects.